### PR TITLE
multi: utilize tx confimation notifications.

### LINKED ISSUE #222: Utilize wallet tx confirmation notifications.
Currently coinbases are being monitored via the `getbalance` rpc, the wallet sync height and waiting a few seconds to avoid spending the coinbase before the wallet is aware of it. This is inefficient and error prone. It'd be ideal to monitor coinbase txs via wallet notifications instead because a received notification confirming outputs of a tx are mature and spendable implicitly also confirm the wallet is aware of the tx as well. In contrast, `gettxout` does not confirm the latter. 

S/O to @jholdstock  for the suggestion.

### PR DESCRIPTION
**Pull request by dnldd**
This uses tx confirmation notifications from the wallet to confirm the coinbases sourced by payments are spendable before attempting to spend them. The readme now  better explains why multiple pool fee address and multiple mining addresses are important to set in production.

Resolves #222 and #218.

### CODE REVIEW

File: pool/paymentmgr.go [hunk 1]
```diff
@@ -904,53 +916,64 @@ func (pm *PaymentMgr) payDividends(ctx context.Context, height uint32) error {
...
+
+				// Remove spendable coinbases from the tx hash set. All
+				// coinbases are spendable when the tx hash set is empty.
+				delete(txHashes, cbHash.String())
+			}
 		}
 
-		if walletHeight >= height {
-			time.Sleep(time.Second * 3)
+		if len(txHashes) == 0 {
```
Comment (jholdstock) [inline pool/paymentmgr.go:959]: Should we close notification stream here?

File: pool/paymentmgr.go [hunk 2]
```diff
@@ -904,53 +916,64 @@ func (pm *PaymentMgr) payDividends(ctx context.Context, height uint32) error {
...
+		maxSpendableHeight = height
 	}
 
-	txB, err := tx.Bytes()
+	hashes := make([]*chainhash.Hash, 0, len(txHashes))
+	for _, hash := range txHashes {
+		hashes = append(hashes, hash)
+	}
+
+	notifSource, err := pm.cfg.GetTxConfNotifications(hashes,
```
Comment (jholdstock) [inline pool/paymentmgr.go:930]: This seems to be creating a new notification stream every time payDividends is called. Are the old streams properly closed?

File: pool/paymentmgr.go (hunk 1 above)
Comment (dnldd) [reply pool/paymentmgr.go:959]: Nope, the stream is terminated at `maxSpendableHeight`.

File: pool/paymentmgr.go (hunk 2 above)
Comment (dnldd) [reply pool/paymentmgr.go:930]: No it's not, each requested stream terminates at the provided `StopAfter` height. `GetTxNotifications` is only returning the function to read the stream with after the confirmation notification request is sent. This function is the `Recv` of the confirmation notification client.

**Comment by jholdstock** [comment]
Tested thoroughly on testnet with 6 mining clients, using both PPS and PPLNS. Performed start/stop testing. Everything is looking good :+1:

**Pull request merged**